### PR TITLE
Fix invalid field types in grouping for kanban layout

### PR DESCRIPTION
**Pull request by licitdev**
Enable selection of certain field types, as `choices` may exist in interface options for `JSON` or `CSV`, such as in checkboxes which does not make sense for the `kanban` layout.

Field types are extracted from the dropdown interface.

https://github.com/directus/directus/blob/cf165e38655265001a6a38dcc69bc8ac1c666789/app/src/interfaces/select-dropdown/index.ts#L11

### CODE REVIEW

**Review by paescuj — APPROVED** [review]
Makes sense!

File: app/src/layouts/kanban/index.ts [hunk 1]
```diff
@@ -72,7 +72,11 @@ export default defineLayout<LayoutOptions, LayoutQuery>({
 				}
 			},
 			group: (field) => {
-				if (field.meta?.options && Object.keys(field.meta?.options).includes('choices')) {
+				if (
+					field.meta?.options &&
+					Object.keys(field.meta?.options).includes('choices') &&
+					['string', 'integer', 'float', 'bigInteger'].includes(field.type)
+				) {
 					return Object.keys(field.meta?.options).includes('choices');
```
Comment (azrikahar) [inline app/src/layouts/kanban/index.ts:80]:
```suggestion
					Object.keys(field.meta.options).includes('choices') &&
					['string', 'integer', 'float', 'bigInteger'].includes(field.type)
				) {
					return Object.keys(field.meta.options).includes('choices');
```

Just a tiny tweak (that should've been on the initial kanban layout PR in hindsight!)

**Review by azrikahar — APPROVED** [review]
>  such as in checkboxes which does not make sense for the kanban layout.

Makes sense 👍 LGTM!

File: app/src/layouts/kanban/index.ts (hunk 1 above)
Comment (licitdev) [reply app/src/layouts/kanban/index.ts]: Ooo nice catch! Thanks! 🚀

**Pull request merged**